tests/host_tools/cargo_build.py

```diff
@@ -28,26 +28,41 @@
 CARGO_UNITTEST_REL_PATH = os.path.join(CARGO_BUILD_REL_PATH, "test")
 
 
+def cargo(
+    subcommand,
+    cargo_args: str = "",
+    subcommand_args: str = "",
+    *,
+    env: dict = None,
+    cwd: str = None,
+):
+    """Executes the specified cargo subcommand"""
+    env = env or {}
+
+    env_string = " ".join(f'{key}="{str(value)}"' for key, value in env.items())
+
+    cmd = f"{env_string} cargo {subcommand} {cargo_args} -- {subcommand_args}"
+
+    if cwd:
+        return utils.run_cmd(cmd, cwd=cwd)
+    return utils.run_cmd(cmd)
```

Comment:
I think if it's None it should work fine unconditionally

Reply:
oh, yeah, you're right! Not sure what I was thinking here, I even remember looking at `run_cmd_sync` thinking "wouldnt this be neat", lol